libethereum/EthereumHost.cpp

```diff
@@ -512,26 +512,32 @@ tuple<vector<shared_ptr<EthereumPeer>>, vector<shared_ptr<EthereumPeer>>, vector
     vector<shared_ptr<EthereumPeer>> chosen;
     vector<shared_ptr<EthereumPeer>> allowed;
     vector<shared_ptr<SessionFace>> sessions;
-
-    size_t peerCount = 0;
+    double percentDecimal = _percent / 100.0;
     foreachPeer([&](std::shared_ptr<EthereumPeer> _p)
     {
         if (_allow(_p.get()))
         {
-            allowed.push_back(_p);
             sessions.push_back(_p->session());
+            allowed.push_back(_p);
         }
-        ++peerCount;
         return true;
     });
 
-    size_t chosenSize = (peerCount * _percent + 99) / 100;
+    if(_percent == 0 || allowed.size() == 0)
+    {
+        return make_tuple(move(chosen), move(allowed), move(sessions));
+    }
+
+    std::random_device randomSeed; 
```

Comment:
Sorry for the long silence, could you please make `random_device` and `mt19937` objects members of the `EthereumHost` class instead of local variables (their initialization might be costly, we don't want to do that each time we call `randomSelection`)